hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java

```diff
@@ -0,0 +1,444 @@
+
+        }
+
+        private CanonicalPath.FeedBuilder newPathPrefix() {
+            return CanonicalPath.of().tenant(tenantId).environment(environmentId).feed(feedId);
+        }
+
+        private BulkPayloadBuilder relationship(CanonicalPath parent, Relationship.Blueprint child) {
+            return relationshipOrEntity(parent.toString(), child, Relationship.class);
+        }
+
+        private BulkPayloadBuilder relationshipOrEntity(String path, AbstractElement.Blueprint newObject,
+                Class<? extends AbstractElement<?, ?>> entityClass) {
+
+            Map<String, List<AbstractElement.Blueprint>> pathEntities = entities.get(path);
+            if (pathEntities == null) {
+                pathEntities = new LinkedHashMap<>();
+                entities.put(path, pathEntities);
+            }
+
+            final String key = toKey(entityClass);
+            List<AbstractElement.Blueprint> list = pathEntities.get(key);
+            if (list == null) {
+                list = new ArrayList<>();
+                pathEntities.put(key, list);
+            }
+            list.add(newObject);
+
+            return this;
+
+        }
+
+        /**
+         * @param resource
+         */
+        public void resource(Resource<?, ?, ?, ?, ?> resource) {
+
+            // get the payload in JSON format
+            org.hawkular.inventory.api.model.Resource.Blueprint rPojo;
+            String resourceTypePath = newPathPrefix().resourceType(getInventoryId(resource.getResourceType())).get()
+                    .toString();
+            rPojo = new org.hawkular.inventory.api.model.Resource.Blueprint(getInventoryId(resource), resourceTypePath,
+                    resource.getProperties());
+
+            StringBuilder parentPath = new StringBuilder();
+            Resource<?, ?, ?, ?, ?> parent = resource.getParent();
+            while (parent != null) {
+                String resourceIdPath = "/" + Util.urlEncode(parent.getID().getIDString());
+                parentPath.insert(0, resourceIdPath);
+                parent = parent.getParent();
+            }
+
+            CanonicalPath parentCanonicalPath = parentPath.length() == 0 ? newPathPrefix().get()
+                    : CanonicalPath.fromPartiallyUntypedString(parentPath.toString(), newPathPrefix().get(),
+                            org.hawkular.inventory.api.model.Resource.class);
+
+            relationshipOrEntity(parentCanonicalPath.toString(), rPojo,
+                    org.hawkular.inventory.api.model.Resource.class);
+
+            String resourcePath = parentPath.toString() + "/" + Util.urlEncode(resource.getID().getIDString());
+            CanonicalPath resourceCanonicalPath = CanonicalPath.fromPartiallyUntypedString(resourcePath,
+                    newPathPrefix().get(), org.hawkular.inventory.api.model.Resource.class);
+
+            // now that the resource is registered, immediately register its configuration
```

Comment:
the resource isn't registered yet - that only happens when inventory is called - I guess the comment just needs to be reworded